fuzzers/libafl-fuzz/src/main.rs

```diff
@@ -0,0 +1,246 @@
+    target_env: Option<HashMap<String, String>>,
+    #[arg(env = "AFL_CYCLE_SCHEDULES")]
+    cycle_schedules: bool,
+    #[arg(env = "AFL_CMPLOG_ONLY_NEW")]
+    cmplog_only_new: bool,
+    #[arg(env = "AFL_PRELOAD")]
+    afl_preload: Option<String>,
+    #[arg(env = "AFL_AUTORESUME")]
+    auto_resume: bool,
+    #[arg(env = "AFL_SKIP_BIN_CHECK")]
+    skip_bin_check: bool,
+    #[arg(env = "AFL_DEFER_FORKSRV")]
+    defer_forkserver: bool,
+    /// in seconds
+    #[arg(env = "AFL_FUZZER_STATS_UPDATE_INTERVAL", default_value = "60")]
+    stats_interval: u64,
+
+    // New Environment Variables
+    #[arg(env = "AFL_NUM_CORES", value_parser = Cores::from_cmdline)]
+    cores: Cores,
+    #[arg(env = "AFL_BROKER_PORT", default_value = "1337")]
+    broker_port: u16,
+
+    // Seed config
+    #[arg(env = "AFL_EXIT_ON_SEED_ISSUES")]
+    exit_on_seed_issues: bool,
+    // renamed from IGNORE_SEED_PROBLEMS
+    #[arg(env = "AFL_IGNORE_SEED_ISSUES")]
+    ignore_seed_issues: bool,
+    #[arg(env = "AFL_CRASHING_SEED_AS_NEW_CRASH")]
+    crash_seed_as_new_crash: bool,
+
+    // TODO:
+    #[arg(env = "AFL_FRIDA_PERSISTENT_ADDR")]
+    frida_persistent_addr: Option<String>,
+    #[arg(env = "AFL_QEMU_CUSTOM_BIN")]
+    qemu_custom_bin: bool,
+    #[arg(env = "AFL_CS_CUSTOM_BIN")]
+    cs_custom_bin: bool,
+    use_wine: bool,
+    uses_asan: bool,
+    frida_mode: bool,
+    qemu_mode: bool,
+    #[cfg(target_os = "linux")]
+    nyx_mode: bool,
+    unicorn_mode: bool,
+    forkserver_cs: bool,
+    no_forkserver: bool,
+    crash_mode: bool,
+    non_instrumented_mode: bool,
+}
+
+const AFL_MAP_SIZE_MIN: usize = usize::pow(2, 3);
+const AFL_MAP_SIZE_MAX: usize = usize::pow(2, 30);
+
+const AFL_DEFAULT_INPUT_LEN_MAX: usize = 1_048_576;
+const AFL_DEFAULT_INPUT_LEN_MIN: usize = 1;
+const OUTPUT_GRACE: u64 = 25;
+
+const PERSIST_SIG: &str = "##SIG_AFL_PERSISTENT##";
+const DEFER_SIG: &str = "##SIG_AFL_DEFER_FORKSRV##";
+const SHMEM_ENV_VAR: &str = "__AFL_SHM_ID";
+
+fn validate_map_size(s: &str) -> Result<usize, String> {
+    let map_size: usize = s
+        .parse()
+        .map_err(|_| format!("`{s}` isn't a valid unsigned integer"))?;
+    if map_size > AFL_MAP_SIZE_MIN && map_size < AFL_MAP_SIZE_MAX {
+        Ok(map_size)
+    } else {
+        Err(format!(
+            "AFL_MAP_SIZE not in range {AFL_MAP_SIZE_MIN} (2 ^ 3) - {AFL_MAP_SIZE_MAX} (2 ^ 30)",
+        ))
+    }
+}
+
+fn validate_harness_input_type(s: &str) -> Result<String, String> {
+    if s != "@@" {
+        return Err("Unknown harness input type. Use \"@@\" for file, omit for stdin ".to_string());
+    }
+    Ok(s.to_string())
+}
+
+/// parse `AFL_TARGET_ENV`; expects: FOO=BAR TEST=ASD
+fn parse_target_env(s: &str) -> Result<Option<HashMap<String, String>>, String> {
+    let env_regex = regex::Regex::new(r"([^\s=]+)\s*=\s*([^\s]+)").unwrap();
+    let mut target_env = HashMap::new();
+    for vars in env_regex.captures_iter(s) {
+        target_env.insert(
```

Comment:
This can probably be an iterator with `.collect()`

Reply:
IMO this is pretty clean considering with iterator closures the error returning gets a bit ugly.